src/theory/quantifiers/single_inv_partition.cpp

```diff
@@ -187,7 +205,17 @@ bool SingleInvocationPartition::init(std::vector<Node>& funcs,
       std::vector<Node> si_subs;
       Trace("si-prt") << "Process conjunct : " << conj[i] << std::endl;
       // do DER on conjunct
-      Node cr = TermUtil::getQuantSimplify(conj[i]);
+      // Must avoid eliminating the first-order input functions. We use a
+      // substitution to avoid this.
```

Comment:
please say which simplification simplification would be eliminating that functions that you are avoiding by substituting with the skolems